Src/DependencyCollector/Shared/Implementation/HttpParsers/AzureQueueHttpParser.cs

```diff
@@ -27,37 +36,30 @@ internal static bool TryParse(ref DependencyTelemetry httpDependency)
                 return false;
             }
 
-            if (!host.EndsWith(".queue.core.windows.net", StringComparison.OrdinalIgnoreCase))
+            if (!HttpParsingHelper.EndsWithAny(host, AzureQueueHostSuffixes))
             {
                 return false;
             }
-
+            
             ////
             //// Queue Service REST API: https://msdn.microsoft.com/en-us/library/azure/dd179423.aspx
             ////
 
             string account = host.Substring(0, host.IndexOf('.'));
 
-            string verb = null;
-            string nameWithoutVerb = name;
+            string verb;
+            string nameWithoutVerb;
 
-            for (int i = 0; i < AzureQueueVerbPrefixes.Length; i++)
-            {
-                var verbPrefix = AzureQueueVerbPrefixes[i];
-                if (name.StartsWith(verbPrefix, StringComparison.OrdinalIgnoreCase))
-                {
-                    verb = name.Substring(0, verbPrefix.Length);
-                    nameWithoutVerb = name.Substring(verbPrefix.Length);
-                    break;
-                }
-            }
+            // try to parse out the verb
+            HttpParsingHelper.ExtractVerb(name, out verb, out nameWithoutVerb, AzureQueueSupportedVerbs);
 
-            var isFirstSlash = nameWithoutVerb[0] == '/' ? 1 : 0;
-            var idx = nameWithoutVerb.IndexOf('/', isFirstSlash); // typically first symbol of the path is '/'
-            string queueName = idx != -1 ? nameWithoutVerb.Substring(isFirstSlash, idx - isFirstSlash) : nameWithoutVerb.Substring(isFirstSlash);
+            List<string> pathTokens = HttpParsingHelper.TokenizeRequestPath(nameWithoutVerb);
+            string queueName = pathTokens.Count > 0 ? pathTokens[0] : string.Empty;
 
             httpDependency.Type = RemoteDependencyConstants.AzureQueue;
-            httpDependency.Name = verb + account + '/' + queueName;
+            httpDependency.Name = string.IsNullOrEmpty(verb)
+                                      ? account + '/' + queueName
+                                      : verb + " " + account + '/' + queueName;
```

Comment:
Minor: Does "+" gives the best performance here with least memory footprint? I do not remember the most performant solution off top of my head but I remember that there were some "gotchas" around string immutability in .NET and Concat() vs. "+" vs. Builder for repetitive operations with similar strings. In particular cases one of them was way faster than others and used less memory.